ui/server.py

```diff
@@ -417,6 +424,63 @@ def update_discovery_group():
         return 'OK', 200
 
 
+def assign_categories(discoveries):
+    """ Add the category to each discovery
+
+    Parameters
+    ----------
+    discoveries: list
+        List of discoveries without assigned categories to them
+
+    Returns
+    -------
+    list
+        List of discoveries with categories assigned to them
+    """
+    rulesdict, cat = _get_rules()
+    for discovery in discoveries:
+        if discovery['rule_id']:
+            discovery['category'] = rulesdict[discovery['rule_id']]['category']
+        else:
+            discovery['category'] = '(rule deleted)'
+
+    return discoveries
+
+
+@app.route('/export_discoveries_csv', methods=['GET', 'POST'])
+def export_discoveries_csv():
+    url = request.form.get('repo_url')
+    _, discoveries = c.get_discoveries(url)
+
+    # Add the category to each discovery
+    discoveries = assign_categories(discoveries)
+
+    # States of discoveries to export
+    states = []
+    if request.form.get('checkAll') == 'all':
+        states = ['new', 'false_positive',
+                  'addressing', 'not_relevant', 'fixed']
+    else:
+        states = request.form.getlist('check')
+
+    # filter out based on states
+    filtered_discoveries = []
+    for d in discoveries:
+        if d['state'] in states:
+            filtered_discoveries.append(d)
+
+    si = io.StringIO()
```

Comment:
what's this? `si` is not a very good name (and it's prone to be confused with `is`)